dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java

```diff
@@ -23,4 +21,45 @@ public Pair(OptionalPair<P, S> pair) {
         this(pair.getFirstValue().orElse(null), pair.getSecondValue().orElse(null));
     }
 
+    public P getFirstValue() {
+        return this.firstValue;
+    }
+
+    public S getSecondValue() {
+        return this.secondValue;
+    }
+
+    public void setFirstValue(P firstValue) {
+        this.firstValue = firstValue;
+    }
+
+    public void setSecondValue(S secondValue) {
+        this.secondValue = secondValue;
+    }
```

Comment:
P and S are nullable, right?

Reply:
They have undefined nullability. Is there a specific annotation I should use or is `@Nullable` fine?

Reply:
These should definitely be nullable though, see line 23 / 21:
` this(pair.getFirstValue().orElse(null), pair.getSecondValue().orElse(null));`
The value `null` is explicitly injected here. I would count this as definitely nullable.